asreview/models/feature_extraction/sbert.py

```diff
@@ -60,19 +61,50 @@ class SBERT(BaseFeatureExtraction):
     transformer_model : str, optional
         The transformer model to use.
         Default: 'all-mpnet-base-v2'
-
+    is_pretrained_SBERT: boolean, optional
+        Default: True
+    pooling_mode: str, optional
+        Pooling mode to get sentence embeddings from word embeddings
+        Default: 'mean'
+        Other options available are 'mean', 'max' and 'cls'.
+        Only used if is_pretrained_SBERT=False
+        mean: Uses mean pooling of word embeddings
+        max: Uses max pooling of word embeddings
+        cls: Uses embeddings of [CLS] token as sentence embeddings
     """
 
     name = "sbert"
     label = "Sentence BERT"
 
-    def __init__(self, *args, transformer_model="all-mpnet-base-v2", **kwargs):
+    def __init__(
+        self,
+        *args,
+        transformer_model="all-mpnet-base-v2",
+        is_pretrained_SBERT=True,
+        pooling_mode="mean",
+        **kwargs
+    ):
         super(SBERT, self).__init__(*args, **kwargs)
         self.transformer_model = transformer_model
+        self.is_pretrained_SBERT = is_pretrained_SBERT
+        self.pooling_mode = pooling_mode
 
     def transform(self, texts):
         _check_st()
 
-        model = SentenceTransformer(self.transformer_model)
+        if self.is_pretrained_SBERT:
+            model = SentenceTransformer(self.transformer_model)
+        else:
+            # If transformer_model is not a pretrained sentence transformer model,
+            # add a pooling layer to get the pooled sentence embeddings from the
+            # word embeddings
+            word_embedding_model = models.Transformer(self.transformer_model)
+            pooling_layer = models.Pooling(
+                word_embedding_model.get_word_embedding_dimension(),
+                pooling_mode=self.pooling_mode,
+            )
+            model = SentenceTransformer(
+                modules=[word_embedding_model, pooling_layer]
+            )
         X = np.array(model.encode(texts))
```

Comment:
Is this casting still needed?

Reply:
I didn't get what you mean by casting here.  Do you mean the keyword argument modules?

Reply:
Sorry, selected wrong line. I am referencing line 109. The `model.encode(texts)` is cast to an array, but I think this is not necessary.

Reply:
You are right. `model.encode(texts)` returns a numpy array and casting is unnecessary.

Reply:
Adding a progress bar will be definitely helpful, especially for extractors taking a long time. But I think rather than using SBERT's progress bar, a reusable "progress bar creator" should be developed which can be used for any other feature extractors such as Doc2vec and fasttextsum

Reply:
Great suggestions. Would be nice to work on this (in another PR)!

Reply:
Maybe something which can be accessed by the front-end so that we can display the progress bar on Webapp as well